src/Umbraco.Web/Editors/ContentController.cs

```diff
@@ -606,7 +606,7 @@ private ContentItemDisplay PostSaveInternal(ContentItemSave contentItem, Func<IC
                 {
                     //ok, so the absolute mandatory data is invalid and it's new, we cannot actually continue!
                     // add the modelstate to the outgoing object and throw a validation message
-                    var forDisplay = MapToDisplay(contentItem.PersistedContent, contentItem.LanguageId);
+                    var forDisplay = MapToDisplay(contentItem.PersistedContent, GetLanguageCulture(contentItem.LanguageId));

```

Comment:
Once this is merged in and merged to my branch, things like this conversion shouldn't be needed since we'll just deal in string everywhere.